src/test/java/seedu/address/logic/commands/CloseClaimCommandTest.java

```diff
@@ -0,0 +1,94 @@
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.client.Client;
+import seedu.address.model.client.exceptions.ClaimException;
+import seedu.address.model.client.exceptions.InsurancePlanException;
+import seedu.address.model.client.insurance.InsurancePlan;
+import seedu.address.model.client.insurance.InsurancePlansManager;
+import seedu.address.model.client.insurance.InvalidInsurancePlan;
+import seedu.address.model.client.insurance.claim.Claim;
+import seedu.address.model.client.insurance.claim.ClaimStub;
+
+class CloseClaimCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    void execute_validIndexUnfilteredList_success() throws CommandException,
+            InsurancePlanException, ClaimException {
+        ModelManager expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+
+        Client clientToEdit = model.getFilteredClientList().get(INDEX_THIRD_CLIENT.getZeroBased());
+        InsurancePlan insurancePlan = clientToEdit.getInsurancePlansManager().getInsurancePlan(0);
+        Claim claim = clientToEdit.getInsurancePlansManager().getInsurancePlan(0).getClaim("A1001");
+
+        CloseClaimCommand closeClaimCommand =
+                new CloseClaimCommand(INDEX_THIRD_CLIENT, 0, claim.getClaimId());
+
+        // make sure all the messages only have client name inside the actual classes
+        // because when the message is created here before modification there can be issues with
+        // the string comparison of the client after deletion and before deletion.
+        String expectedMessage = String.format(CloseClaimCommand.MESSAGE_CLOSE_CLAIM_SUCCESS,
+                clientToEdit.getName().toString(), insurancePlan, claim.getClaimId());
+
+        // rebuilds a copy of the claims from string to prevent same-object manipulation
+        String insurancePlanString = insurancePlan.toString();
+        String claimsString = clientToEdit.getInsurancePlansManager().convertClaimsToJson();
+        InsurancePlansManager updatedInsurancePlanManager = new InsurancePlansManager(insurancePlanString);
+        updatedInsurancePlanManager.addAllClaimsFromJson(claimsString);
+        updatedInsurancePlanManager.closeClaim(insurancePlan, claim);
+
+        Client updatedClient = new Client(clientToEdit.getName(), clientToEdit.getPhone(), clientToEdit.getEmail(),
+                clientToEdit.getAddress(), updatedInsurancePlanManager, clientToEdit.getTags());
+
+        expectedModel.setClient(clientToEdit, updatedClient);
+
+        assertCommandSuccess(closeClaimCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    void execute_invalidClaim_exceptionHandled() {
```

Comment:
Instead of naming it as `execute_invalidClaim_exceptionHandled`, the method can be named as `execute_invalidClaim_throwsCommandException`. This would give an intuitive understanding of the test case without reading into the implementation details

Reply:
yep that makes more sense, will change!